Fundsoft/report.md

```diff
@@ -43,3 +43,25 @@ product are very good and the product works as intended.
     It runs excellently. With single scripts `pdm test` and `pdm coverage`, the tests and test coverage is executed with `pytest`.
 
 
+## Overview of the architecture and purpose of pdm
+
+The purpose of pdm is to provide a framework for python projects that handles package dependencies automatically. It also resolves the version requirements for all dependencies. The user specifies what version of what library that the project needs in a `pyproject.toml` file or by adding it with commands from the command line. Unlike other Python packaging tools, like Poetry and Hatch, pdm is not tightly integrated into a built-in build backend. Instead it uses PEP 517, a build-system independent format. pdm also allows the user to customize it with plugins.
+
+<div style="text-align:center;">
+    <figure>
+        <img src="img/architecture-diagram.png" alt="Diagram of the architecture" width="500"/>
+        <figcaption>Diagram of the architecture of pdm.</figcaption>
+    </figure>
+</div>
+
+
+**Key functions in pdm source code we described in the report for assigment 3:**
```

Comment:
Instead of introducing the functions first you could introduce the concepts first and then the functions. So for instance, instead of: "`do_lock`: locking in a package manager means locking down the versions of the dependencies that are used in the project. `do_lock` does that locking as well as updating the lock file. " it could be: "Locking in a package manager means locking down the versions of the dependencies that are used in the project. In PDM, it is the function `do_lock` that does this and it works by bla bla bla